openedx/core/djangoapps/cookie_metadata/middleware.py

```diff
@@ -0,0 +1,59 @@
+"""Middleware to changes name of an incoming cookie"""
+from django.conf import settings
+
+
+class CookieNameChange:
+    """Changes name of an incoming cookie"""
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        """
+        Changes the names of a cookie in request.COOKIES
+
+        For this middleware to run:
+        - set COOKIE_NAME_CHANGE_ACTIVATE_EXPAND_PHASE =  True
+        - COOKIE_NAME_CHANGE_EXPAND_INFO is a dict and has following info:
+            "old_name": Previous name of cookie
+            "new_name": New name of cookie
+            if you also want to delete the cookie in user's browser, set "old_domain" as well
+
+        Actions taken by middleware:
+            - will delete any cookie with "old_name"
+            - if create a new cookie with "new_name" and set its value to value of "old_name" cookie
+                + it will not modify cookie with "new_name" if it already exists in request.COOKIES
+        """
+
+        # .. toggle_name: COOKIE_NAME_CHANGE_ACTIVATE_EXPAND_PHASE
+        # .. toggle_implementation: SettingToggle
```

Comment:
Looks like a `DjangoSetting` instead -- but I would support changing it to a SettingToggle. :-)

Reply:
I decided to just keep it as a DjangoSetting and fixed the doc.